[Vrf] test_vrf fails due to missing attributes in topo_t0.yml

Description: 
After running test_vrf some of test cases fail with such error: 
"AnsibleUndefinedVariable: 'dict object' has no attribute 'podset_number'". 
The reason is that these attributes were deleted from topo_t0.yml by commit #1233 
```
    podset_number: 200
    tor_number: 16
    tor_subnet_number: 2
    max_tor_subnet_number: 16
    tor_subnet_size: 128
    spine_asn: 65534
    leaf_asn_start: 64600
    tor_asn_start: 65100
    failure_rate: 0
```
These attributes is mainly used in test_vrf by vrf_fib.j2


**Steps to reproduce the issue:**
1. Deploy t0 topology.
2. Deploy minigrapgh on DUT
3. Run test_vrf.py 

**Describe the results you received:**
```
E           RunAnsibleModuleFail: run module template failed, Ansible Results =>
E           {"changed": false, "failed": true, "msg": "AnsibleUndefinedVariable: 'dict object' has no attribute 'podset_number'"}

vrf/test_vrf.py::TestVrfIsolation::test_neigh_isolate_vrf1_from_vrf2 ERROR                                                                                       
vrf/test_vrf.py::TestVrfIsolation::test_neigh_isolate_vrf2_from_vrf1 ERROR                                                                                       
vrf/test_vrf.py::TestVrfIsolation::test_fib_isolate_vrf1_from_vrf2 ERROR                                                                                        
vrf/test_vrf.py::TestVrfIsolation::test_fib_isolate_vrf2_from_vrf1 ERROR
```

**Describe the results you expected:**
```
vrf/test_vrf.py::TestVrfIsolation::test_neigh_isolate_vrf1_from_vrf2 PASSED
vrf/test_vrf.py::TestVrfIsolation::test_neigh_isolate_vrf2_from_vrf1 PASSED
vrf/test_vrf.py::TestVrfIsolation::test_fib_isolate_vrf1_from_vrf2 PASSED
vrf/test_vrf.py::TestVrfIsolation::test_fib_isolate_vrf2_from_vrf1 PASSED
```

**Additional information you deem important:**
<!--
software/ASIC/Hardware Platform version and info
-->
    **Output of `show version`:**

    ```
    SONiC Software Version: SONiC.master.22-dirty-20201030.044825
    Distribution: Debian 10.6
    Kernel: 4.19.0-9-2-amd64
    Build commit: b132ca09
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

--- a/ansible/vars/topo_t0-116.yml
+++ b/ansible/vars/topo_t0-116.yml
@@ -162,6 +162,11 @@ configuration_properties:
     dut_asn: 4200065100
     dut_type: ToRRouter
     swrole: leaf
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
     spine_asn: 65534
     leaf_asn_start: 4200064600
     tor_asn_start: 4200065500

--- a/ansible/vars/topo_t0.yml
+++ b/ansible/vars/topo_t0.yml
@@ -81,6 +81,15 @@ configuration_properties:
     swrole: leaf
     nhipv4: 10.10.246.254
     nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
+    spine_asn: 65534
+    leaf_asn_start: 64600
+    tor_asn_start: 65500
+    failure_rate: 0
 
 configuration:
   ARISTA01T1:

--- a/ansible/vars/topo_t1-64-lag-clet.yml
+++ b/ansible/vars/topo_t1-64-lag-clet.yml
@@ -103,6 +103,11 @@ configuration_properties:
     dut_type: LeafRouter
     nhipv4: 10.10.246.100
     nhipv6: FC0A::C9
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
   spine:
     swrole: spine
     podset_number: 200

--- a/ansible/vars/topo_t1-64-lag.yml
+++ b/ansible/vars/topo_t1-64-lag.yml
@@ -107,6 +107,11 @@ configuration_properties:
     dut_type: LeafRouter
     nhipv4: 10.10.246.254
     nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
   spine:
     swrole: spine
   tor:

--- a/ansible/vars/topo_t1-64.yml
+++ b/ansible/vars/topo_t1-64.yml
@@ -263,6 +263,11 @@ configuration_properties:
     dut_type: LeafRouter
     nhipv4: 10.10.246.254
     nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
   spine:
     swrole: spine
   tor:

--- a/ansible/vars/topo_t1-lag.yml
+++ b/ansible/vars/topo_t1-lag.yml
@@ -111,6 +111,11 @@ configuration_properties:
     dut_type: LeafRouter
     nhipv4: 10.10.246.254
     nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
   spine:
     swrole: spine
   tor:

--- a/ansible/vars/topo_t1.yml
+++ b/ansible/vars/topo_t1.yml
@@ -135,6 +135,11 @@ configuration_properties:
     dut_type: LeafRouter
     nhipv4: 10.10.246.254
     nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
   spine:
     swrole: spine
   tor:

--- a/tests/common/plugins/fib.py
+++ b/tests/common/plugins/fib.py
@@ -27,7 +27,7 @@ def announce_routes(ptfip, port, routes):
 def generate_routes(family, podset_number, tor_number, tor_subnet_number,
                     spine_asn, leaf_asn_start, tor_asn_start,
                     nexthop, nexthop_v6,
-                    tor_subnet_size = 128, max_tor_subnet_number = 16,
+                    tor_subnet_size, max_tor_subnet_number,
                     router_type = "leaf", tor_index=None):
     routes = []
 
@@ -99,11 +99,14 @@ def generate_routes(family, podset_number, tor_number, tor_subnet_number,
 def fib_t0(ptfhost, tbinfo, localhost, topology=None):
     logger.info("use fib_t0 to setup routes for topo {}".format(tbinfo['topo']['name']))
 
-    podset_number = 200
-    tor_number = 16
-    tor_subnet_number = 2
-
     common_config_topo = tbinfo['topo']['properties']['configuration_properties']['common']
+    podset_number = common_config_topo.get("podset_number", 200)
+    tor_number = common_config_topo.get("tor_number", 16)
+    tor_subnet_number = common_config_topo.get("tor_subnet_number", 2)
+    max_tor_subnet_number = common_config_topo.get("max_tor_subnet_number", 16)
+    tor_subnet_size = common_config_topo.get("tor_subnet_size", 128)
+    nhipv4 = common_config_topo.get("nhipv4", "10.10.246.254")
+    nhipv6 = common_config_topo.get("nhipv6", "fc0a::ff")
     spine_asn = common_config_topo.get("spine_asn", 65534)
     leaf_asn_start = common_config_topo.get("leaf_asn_start", 64600)
     tor_asn_start = common_config_topo.get("tor_asn_start", 65500)
@@ -111,8 +114,8 @@ def fib_t0(ptfhost, tbinfo, localhost, topology=None):
     ptf_hostname = tbinfo['ptf']
     ptfip = ptfhost.host.options['inventory_manager'].get_host(ptf_hostname).vars['ansible_host']
 
-    local_ip = ipaddress.IPAddress("10.10.246.254")
-    local_ipv6 = ipaddress.IPAddress("fc0a::ff")
+    local_ip = ipaddress.IPAddress(nhipv4)
+    local_ipv6 = ipaddress.IPAddress(nhipv6)
     for k, v in tbinfo['topo']['properties']['configuration'].items():
         vm_offset = tbinfo['topo']['properties']['topology']['VMs'][k]['vm_offset']
         peer_ip = ipaddress.IPNetwork(v['bp_interface']['ipv4'])
@@ -155,10 +158,10 @@ def fib_t0(ptfhost, tbinfo, localhost, topology=None):
 
         routes_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
                                     spine_asn, leaf_asn_start, tor_asn_start,
-                                    local_ip, local_ipv6)
+                                    local_ip, local_ipv6, tor_subnet_size, max_tor_subnet_number)
         routes_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
                                     spine_asn, leaf_asn_start, tor_asn_start,
-                                    local_ip, local_ipv6)
+                                    local_ip, local_ipv6, tor_subnet_size, max_tor_subnet_number)
 
         announce_routes(ptfip, port, routes_v4)
         announce_routes(ptfip, port6, routes_v6)
@@ -167,18 +170,22 @@ def fib_t0(ptfhost, tbinfo, localhost, topology=None):
 def fib_t1_lag(ptfhost, tbinfo, localhost):
     logger.info("use fib_t1_lag to setup routes for topo {}".format(tbinfo['topo']['name']))
 
-    podset_number = 200
-    tor_number = 16
-    tor_subnet_number = 2
-
-    leaf_asn_start  = 64600
-    tor_asn_start   = 65500
+    common_config_topo = tbinfo['topo']['properties']['configuration_properties']['common']
+    podset_number = common_config_topo.get("podset_number", 200)
+    tor_number = common_config_topo.get("tor_number", 16)
+    tor_subnet_number = common_config_topo.get("tor_subnet_number", 2)
+    max_tor_subnet_number = common_config_topo.get("max_tor_subnet_number", 16)
+    tor_subnet_size = common_config_topo.get("tor_subnet_size", 128)
+    nhipv4 = common_config_topo.get("nhipv4", "10.10.246.254")
+    nhipv6 = common_config_topo.get("nhipv6", "fc0a::ff")
+    leaf_asn_start = common_config_topo.get("leaf_asn_start", 64600)
+    tor_asn_start = common_config_topo.get("tor_asn_start", 65500)
 
     ptf_hostname = tbinfo['ptf']
     ptfip = ptfhost.host.options['inventory_manager'].get_host(ptf_hostname).vars['ansible_host']
 
-    local_ip = ipaddress.IPAddress("10.10.246.254")
-    local_ipv6 = ipaddress.IPAddress("fc0a::ff")
+    local_ip = ipaddress.IPAddress(nhipv4)
+    local_ipv6 = ipaddress.IPAddress(nhipv6)
 
     for k, v in tbinfo['topo']['properties']['configuration'].items():
         vm_offset = tbinfo['topo']['properties']['topology']['VMs'][k]['vm_offset']
@@ -231,10 +238,12 @@ def fib_t1_lag(ptfhost, tbinfo, localhost):
         if router_type:
             routes_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
                                         None, leaf_asn_start, tor_asn_start,
-                                        local_ip, local_ipv6, router_type=router_type, tor_index=tor_index)
+                                        local_ip, local_ipv6, tor_subnet_size, max_tor_subnet_number,
+                                        router_type=router_type, tor_index=tor_index)
             routes_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
                                         None, leaf_asn_start, tor_asn_start,
-                                        local_ip, local_ipv6, router_type=router_type, tor_index=tor_index)
+                                        local_ip, local_ipv6, tor_subnet_size, max_tor_subnet_number,
+                                        router_type=router_type, tor_index=tor_index)
             announce_routes(ptfip, port, routes_v4)
             announce_routes(ptfip, port6, routes_v6)
 

--- a/tests/decap/test_decap.py
+++ b/tests/decap/test_decap.py
@@ -64,28 +64,28 @@ def gen_fib_info(ptfhost, tbinfo, cfg_facts):
     downlink_ports_str = " ".join(get_downlink_ports(topology, topo_type))
 
     fibs = []
-
+    common_config_topo = tbinfo['topo']['properties']['configuration_properties']['common']
     podset_number = 5  # Limit the number of podsets to limit test execution time
-    tor_number = 16
-    tor_subnet_number = 2
+    tor_number = common_config_topo.get("tor_number", 16)
+    tor_subnet_number = common_config_topo.get("tor_subnet_number", 2)
+    max_tor_subnet_number = common_config_topo.get("max_tor_subnet_number", 16)
+    tor_subnet_size = common_config_topo.get("tor_subnet_size", 128)
 
     # routes to uplink
     routes_uplink_v4 = []
     routes_uplink_v6 = []
     if topo_type == "t0":
         routes_uplink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
-                                            0, 0, 0,
-                                            "", "")
+                                            0, 0, 0, "", "", tor_subnet_size, max_tor_subnet_number)
         routes_uplink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
-                                            0, 0, 0,
-                                            "", "")
+                                            0, 0, 0, "", "", tor_subnet_size, max_tor_subnet_number)
     elif topo_type == "t1":
         routes_uplink_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
-                                            0, 0, 0,
-                                            "", "", router_type="spine")
+                                            0, 0, 0, "", "", tor_subnet_size, max_tor_subnet_number,
+                                            router_type="spine")
         routes_uplink_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
-                                            0, 0, 0,
-                                            "", "", router_type="spine")
+                                            0, 0, 0, "", "", tor_subnet_size, max_tor_subnet_number,
+                                            router_type="spine")
 
     for prefix, _, _ in routes_uplink_v4:
         fibs.append("{} {}".format(prefix, uplink_ports_str))
@@ -97,12 +97,12 @@ def gen_fib_info(ptfhost, tbinfo, cfg_facts):
     if topo_type == "t1":
         for tor_index in range(tor_number):
             routes_downlink_v4.extend(generate_routes("v4", podset_number, tor_number, tor_subnet_number,
-                                      0, 0, 0,
-                                      "", "", router_type="tor", tor_index=tor_index))
+                                      0, 0, 0, "", "", tor_subnet_size, max_tor_subnet_number,
+                                      router_type="tor", tor_index=tor_index))
 
             routes_downlink_v6.extend(generate_routes("v6", podset_number, tor_number, tor_subnet_number,
-                                      0, 0, 0,
-                                      "", "", router_type="tor", tor_index=tor_index))
+                                      0, 0, 0, "", "", tor_subnet_size, max_tor_subnet_number,
+                                      router_type="tor", tor_index=tor_index))
 
     for prefix, _, _ in routes_downlink_v4:
         fibs.append("{} {}".format(prefix, downlink_ports_str))

